build.sbt

```diff
@@ -207,3 +231,54 @@ lazy val algebirdSpark = module("spark").settings(
   libraryDependencies += "org.apache.spark" %% "spark-core" % "1.3.0" % "provided"
 ).dependsOn(algebirdCore, algebirdTest % "test->test")
 
+lazy val docsMappingsAPIDir = settingKey[String]("Name of subdirectory in site target directory for api docs")
+
+lazy val docSettings = Seq(
+  micrositeName := "Algebird",
+  micrositeDescription := "Abstract Algebra for Scala.",
+  micrositeAuthor := "Algebird's contributors",
+  micrositeHighlightTheme := "atom-one-light",
+  micrositeHomepage := "http://twitter.github.io/algebird",
+  micrositeBaseUrl := "algebird",
+  micrositeDocumentationUrl := "api",
+  micrositeGithubOwner := "twitter",
+  micrositeExtraMdFiles := Map(file("CONTRIBUTING.md") -> "contributing.md"),
+  micrositeGithubRepo := "algebird",
+  micrositePalette := Map(
+    "brand-primary" -> "#5B5988",
+    "brand-secondary" -> "#292E53",
+    "brand-tertiary" -> "#222749",
+    "gray-dark" -> "#49494B",
+    "gray" -> "#7B7B7E",
+    "gray-light" -> "#E5E5E6",
+    "gray-lighter" -> "#F4F3F4",
+    "white-color" -> "#FFFFFF"),
+  autoAPIMappings := true,
+  unidocProjectFilter in (ScalaUnidoc, unidoc) :=
+    inProjects(docsSourcesAndProjects(scalaVersion.value)._2:_*),
+  docsMappingsAPIDir := "api",
+  addMappingsToSiteDir(mappings in (ScalaUnidoc, packageDoc), docsMappingsAPIDir),
+  ghpagesNoJekyll := false,
+  fork in tut := true,
+  fork in (ScalaUnidoc, unidoc) := true,
+  scalacOptions in (ScalaUnidoc, unidoc) ++= Seq(
+    "-doc-source-url", "https://github.com/twitter/algebird/tree/develop€{FILE_PATH}.scala",
+    "-sourcepath", baseDirectory.in(LocalRootProject).value.getAbsolutePath,
+    "-diagrams"
+  ),
+  git.remoteRepo := "git@github.com:twitter/algebird.git",
+  includeFilter in makeSite := "*.html" | "*.css" | "*.png" | "*.jpg" | "*.gif" | "*.js" | "*.swf" | "*.yml" | "*.md"
+)
+
+// Documentation is generated for projects defined in
+// `docsSourcesAndProjects`.
+lazy val docs = project
+  .enablePlugins(MicrositesPlugin)
+  .settings(moduleName := "algebird-docs")
+  .settings(sharedSettings)
+  .settings(noPublishSettings)
+  .settings(unidocSettings)
+  .settings(ghpages.settings)
+  .settings(docSettings)
+  .settings(tutScalacOptions ~= (_.filterNot(Set("-Ywarn-unused-import", "-Ywarn-dead-code"))))
+  .dependsOn(algebirdCore)
```

Comment:
for now we only depend on core to generate tutorial stuff... users can add if they want to write docs for the non-core projects.